docs/sources/setup/upgrade/upgrade-from-2x/index.md

```diff
@@ -47,6 +47,28 @@ helm upgrade loki grafana/loki \
 
 You will need to manually delete the existing stateful set for the above command to work.
 
+#### Notable changes
+
+The `grafana/loki` used `Secret` as storage for configuration, you can set `.loki.existingSecretForConfig` to continue using secret or migrate your configuration to `ConfigMap`. Specifying the loki config in `values.yaml` is still available. In the old chart it was under `.config`, the new chart allows specifying either `.loki.config` or `.loki.structuredConfig` which takes precedence over the former alternative.
+
+Similarly when using `extraVolumes`, the configuration is now nested under `.singleBinary.extraVolumes` or `.read.extraVolumes` + `.write.extraVolumes` if you decide to migrate to loki scalable mode.
+
+#### Dependencies
+
+The `grafana/loki` chart was used to install purely **loki**. New chart since `v3.x` also bundles two dependencies - **minio** and **grafana-agent-operator**. If you have already installed any of these tools independently and wish to continue managing them that way, you can explicitly disable these dependencies in your `values.yaml`:
```

Comment:
```suggestion
The `grafana/loki` chart was only used to install Loki. New charts since `v3.x` also bundle two dependencies - **minio** and **grafana-agent-operator**. If you have already installed either of these independently and wish to continue managing them separately, you can explicitly disable these dependencies in your `values.yaml` as shown in the following examples:
```